README.md

```diff
@@ -37,143 +37,160 @@ pre-commit install
+password = "your_password"  # Replace this value with your password on the PASQAL platform.
 ```
 
-Once you have serialized your sequence, you can send it with the SDK with the following code
+Ideally, do not write this password in a script but provide in through the command-line or as a secret environment variable.
 
-```python
-from pasqal_cloud import SDK
-
-project_id="your_project_id" # Replace this value by your project_id on the PASQAL platform.
-username="your_username" # Replace this value by your username or email on the PASQAL platform.
-password="your_password" # Replace this value by your password on the PASQAL platform.
+#### Method 1: Username + Password
+If you know your credentials, you can pass them to the SDK instance on creation.
 
-# Initialize the cloud client
+```python
 sdk = SDK(username=username, password=password, project_id=project_id)
+```
 
-# When creating a job, select a number of runs and set the desired values for the variables
-# defined in the sequence
-job1 = {"runs": 20, "variables": {"omega_max": 6} }
-job2 = {"runs": 50, "variables": {"omega_max": 10.5} }
+#### Method 2: Username only
+If you only want to insert your username, but want a solution to have your password being secret,
+you can run the SDK without a password. A prompt will then ask for your password
 
-# Send the batch of jobs to the QPU and wait for the results
-batch = sdk.create_batch(serialized_sequence, [job1,job2], wait=True)
+```python
+sdk = SDK(username=username, project_id=project_id)
+```
 
-# You can also choose to run your batch on an emulator using the optional argument 'emulator'
-# For using a basic single-threaded QPU emulator that can go up to 10 qubits, you can specify the "EMU_FREE" emulator.
-from pasqal_cloud.device import EmulatorType
-batch = sdk.create_batch(serialized_sequence, [job1,job2], emulator=EmulatorType.EMU_FREE)
+#### Method 3: Use a custom token provider
+You can define a custom class to provide the token.
+For example, if you know your token, you can use that token to authenticate directly to our APIs as follows.
 
-# Once the QPU has returned the results, you can access them with the following:
-for job in batch.ordered_jobs:
-    print(job.result)
+```python
+class CustomTokenProvider(TokenProvider):
+    def get_token(self):
+        return "your-token" # Replace this value with your token
 
+
+sdk = SDK(token_provider=CustomTokenProvider(), project_id=project_id)
 ```
+Alternatively, create a custom TokenProvider that inherits from ExpiringTokenProvider. You should define a
+custom _query_token method which fetches your token. See Auth0TokenProvider implementation for an example.
 
-### Workload Creation
+### Create a batch of jobs
+
+The package main component is a python object called `SDK` which can be used to create a `Batch` and send it automatically
+to Pasqal APIs using an API token generated in the [user portal](https://portal.pasqal.cloud).
```

Comment:
I'm not sure what this API token refers to. This is reminiscent of API keys but these have been deleted a year ago. I would suggest removing the reference to this token, unless I forgot something

Reply:
Now you'll need the project ID though, so perhaps worth referencing that if not done elsewhere

Reply:
it is referenced above in the authentication part